transform/all_collection/spark/utils/common_df_transformations.py

```diff
@@ -0,0 +1,246 @@
+            if authors:
+                for author in authors:
+                    # Fullname
+                    author_names_row.append(author["fullname"])
+                    # Pids
+                    if author["pid"]:
+                        author_pid = [
+                            [author["fullname"]],
+                            author["pid"]["id"]["value"],
+                        ]
+                    else:
+                        author_pid = [[author["fullname"]], None]
+                    author_pids_row.append(author_pid)
+
+            authors_names_column.append(author_names_row)
+            authors_pids_column.append(author_pids_row)
+
+    harvested_properties[AUTHOR_NAMES] = authors_names_column
+    harvested_properties[AUTHOR_PIDS] = authors_pids_column
+
+
+def check_type(df: DataFrame, desired_type: str) -> None:
+    """Check if all records have the right type"""
+    df_type = df.select(TYPE).collect()
+    assert all(
+        (row[TYPE] == desired_type for row in df_type)
+    ), f"Not all records have {TYPE}: {desired_type}"
+
+
+def harvest_sdg(df: DataFrame, harvested_properties: Dict) -> None:
+    """Harvest sdg from subjects"""
+    subjects = df.select("subject").collect()
+    sdg_column = []
+
+    for subject in subjects:
+        sdg = subject["subject"]["sdg"]
+        sdg_list = []
+        if sdg:
+            for value in sdg:
+                sdg_list.append(value["value"])
+            sdg_column.append(sdg_list)
+        else:
+            sdg_column.append(None)
+
+    harvested_properties[SDG] = sdg_column
+
+
+def harvest_fos(df: DataFrame, harvested_properties: Dict) -> None:
+    """Harvest fos from subjects"""
+    subjects = df.select("subject").collect()
+    fos_column = []
+
+    for subject in subjects:
+        fos = subject["subject"]["fos"]
+        fos_list = []
+        if fos:
+            for value in fos:
+                fos_list.append(value["value"])
+            fos_column.append(fos_list)
+        else:
+            fos_column.append(None)
+
+    harvested_properties[FOS] = fos_column
+
+
+def simplify_bestaccessright(df: DataFrame) -> DataFrame:
+    """Simplify best_access_right - get only label and convert structure to a string"""
+    return df.withColumn("bestaccessright", col("bestaccessright")["label"])
+
+
+def simplify_language(df: DataFrame) -> DataFrame:
+    """Simplify language - get only label and convert structure to a string"""
+    return df.withColumn("language", col("language")["label"])
+
+
+def create_open_access(
+    df: DataFrame, harvested_properties: Dict, col_name: str
+) -> None:
+    """Create boolean value whether record is open access or not, based on col_name"""
+    best_access_right = df.select(col_name).collect()
+    open_access_column = []
+
+    for access in best_access_right:
+        if access[col_name] in {
+            "OPEN",
+            "open access",
+            "fully_open_access",
+            "open_access",
```

Comment:
In UI I've done it on regex /open.(access)?/gi

Reply:
My approach is more bulletproof. Some records may have `open` in the `best_access_right` and do not be `open access.` I will leave it as it is